datahub-web-react/src/app/entity/shared/containers/profile/header/EntityHeader.tsx

```diff
@@ -69,8 +69,20 @@ const ExternalUrlButton = styled(Button)`
     padding-right: 12px;
 `;
 
-export function getCanEditName(entityType: EntityType, privileges?: PlatformPrivileges) {
+export function getCanEditName(
+    entityType: EntityType,
+    privileges?: PlatformPrivileges,
+    me?: AuthenticatedUser,
+    entityData?: GenericEntityProperties,
+) {
     switch (entityType) {
+        case EntityType.Dataset: {
```

Comment:
I definitely think that we should avoid hardcoding in whether the current user is able to edit the name..

This requires a way to ask our backend whether the current user is able to change the name. Likely we'll need a new Metadata Privilege called "Update Entity Name" that will be able to represent this. Then we'll need to fetch those privileges from the frontend

Reply:
This is a specific need that we have on our end wherein only the owners must be able to edit certain fields within  the UI. Even if we had a Metadata Privilege called "Update Entity Name", we would still need something to prevent updates on datasets that one does not own. If it makes sense to not have it in the UI, I will remove this logic and leave it similar to how the description is.

Reply:
I have added a new privilege

> > Hey @jaykadambi are you still working on this PR?
> 
> @aditya-radhakrishnan - Yes. I had some questions/comments back to @jjoyce0510. Perhaps I can go ahead and commit the updated code and let @jjoyce0510 review when he gets a chance. Thanks!

@jjoyce0510 - I have made the next set of commits to address the PR comments. Please review and let me know your feedback. Thanks!

cc - @aditya-radhakrishnan.